src/dashboards/components/AutoRefreshOverlay.tsx

```diff
@@ -97,6 +110,39 @@ const AutoRefreshOverlay: FC = () => {
     'Hours'
   )
 
+  const refreshOptions = [
+    {
+      id: 'indefinite-auto-refresh',
+      title: 'Indefinite',
+      active: infiniteDuration,
+    },
+    {
+      id: 'custom-auto-refresh',
+      title: 'Custom',
+      active: !infiniteDuration,
+    },
+  ]
+
+  const isValidTimeoutInput = (input: string) => {
+    if (+input <= 0) {
+      return false
+    }
+    const unit = inactivityTimeoutCategory
+    let isValid = false
+    switch (unit) {
+      case TIMEOUT_CATEGORY.Minutes:
+        isValid = +input < MAX_MINUTE
+        break
+      case TIMEOUT_CATEGORY.Hours:
+        isValid = +input < MAX_HOUR
+        break
+      case TIMEOUT_CATEGORY.Days:
+        isValid = +input < MAX_DAY
+        break
+    }
+    return input === 'Never' || isValid
```

Comment:
this all checks out, but it makes for a confusing function where we check for a negative value of a number coerced string where the string can be `Never`. Can we add a condition in the beginning to handle the `Never` case upfront, that way the rest of the function flows with respect to the assumption that the string should be a number from thereon?